### PR TITLE
Upgrade/django2 on_delete

### PR DESCRIPTION
**Pull request by Edwinmore**

### CODE REVIEW

File: djangoplicity/pages/migrations/0001_initial.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ class Migration(migrations.Migration):
             fields=[
                 ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
                 ('url', models.CharField(help_text="Example: '/about/contact/'. Make sure to have leading and trailing slashes. Good and descriptive URLs are important for good user experience and search engine ranking.", unique=True, max_length=200, verbose_name='URL', db_index=True)),
-                ('page', models.ForeignKey(to='pages.Page', null=True)),
+                ('page', models.ForeignKey(to='pages.Page', null=True, on_delete=models.deletion.CASCADE)),
```
Comment (EdisonArango) [inline djangoplicity/pages/migrations/0001_initial.py]: You are missing the import:
```python
import django.db.models.deletion
````

File: djangoplicity/releases/migrations/0001_initial.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ class Migration(migrations.Migration):
                 ('last_modified', models.DateTimeField(auto_now=True, verbose_name='Last modified')),
                 ('created', models.DateTimeField(auto_now_add=True, verbose_name='Created')),
                 ('facility', djangoplicity.metadata.translation.fields.TranslationFacilityManyToManyField(help_text='Telescopes or observatories used for the observations.', to='metadata.Facility', null=True, verbose_name='Facility', blank=True)),
-                ('kids_image', models.ForeignKey(related_name='kids_image_release_set', blank=True, to='media.Image', help_text='Use this to override the default Release image.', null=True)),
+                ('kids_image', models.ForeignKey(related_name='kids_image_release_set', blank=True, to='media.Image', help_text='Use this to override the default Release image.', null=True, on_delete=models.deletion.CASCADE)),
```
Comment (EdisonArango) [inline djangoplicity/releases/migrations/0001_initial.py]: You are missing the import:
```python
import django.db.models.deletion
```

File: djangoplicity/archives/contrib/info/defaults.py [hunk 3]
```diff
@@ -21,6 +20,11 @@
 
 __all__ = ('categories', 'admin_edit', 'admin_edit_for_site', 'admin_add_translation', 'published', 'featured', 'boolean_property', 'priority', 'release_date', 'paper_size', 'link_field')
 
+import django
+if django.VERSION >= (2, 2):
```
Comment (fraroca97) [inline djangoplicity/archives/contrib/info/defaults.py]: `django.core.urlresolvers` was removed in django 2.0, therefore the current statement will raise an exception when used in django [2.0, 2.2) range.

File: djangoplicity/archives/contrib/satchmo/freeorder/models.py [hunk 4]
```diff
@@ -23,6 +22,12 @@
 import string
 import random
 
+import django
+if django.VERSION >= (2, 2):
```
Comment (fraroca97) [inline djangoplicity/archives/contrib/satchmo/freeorder/models.py]: Same comment as before, `django.core.urlresolvers` was removed in 2.0.

File: djangoplicity/reports/models.py [hunk 5]
```diff
@@ -6,10 +6,14 @@
 #   Luis Clara Gomes <lcgomes@eso.org>
 #
 from django.db import models
-from django.core.urlresolvers import reverse
 from django.utils.translation import ugettext_lazy as _
 from django.utils.encoding import python_2_unicode_compatible
 
+import django
+if django.VERSION >= (2, 2):
```
Comment (fraroca97) [inline djangoplicity/reports/models.py]: Same comment as before, `django.core.urlresolvers` was removed in django 2.0.

File: djangoplicity/science/migrations/0001_initial.py [hunk 6]
```diff
@@ -47,8 +48,8 @@ class Migration(migrations.Migration):
                 ('main_visual', models.BooleanField(default=False)),
                 ('override_id', models.SlugField(null=True, verbose_name='Override ID', blank=True)),
                 ('hide', models.BooleanField(default=False, verbose_name='Hide on kiosk')),
-                ('archive_item', models.ForeignKey(verbose_name='Related Image', to='media.Image')),
-                ('science_announcement', models.ForeignKey(verbose_name='Related science announcement', to='science.ScienceAnnouncement')),
+                ('archive_item', models.ForeignKey(verbose_name='Related Image', to='media.Image', on_delete=django.db.deletion.CASCADE)),
```
Comment (fraroca97) [inline djangoplicity/science/migrations/0001_initial.py]: Module should be `django.db.models.deletion`, currently the `models` module is missing.

File: djangoplicity/science/models.py [hunk 7]
```diff
@@ -42,6 +41,11 @@
 from djangoplicity.media.models import Image
 from djangoplicity.translation.models import TranslationModel
 
+import django
+if django.VERSION >= (2, 2):
```
Comment (fraroca97) [inline djangoplicity/science/models.py]: Same comment as before, `django.core.urlresolvers` was removed in django 2.0.

File: djangoplicity/archives/contrib/satchmo/freeorder/models.py (hunk 4 above)
Comment (EdisonArango) [reply djangoplicity/archives/contrib/satchmo/freeorder/models.py]: @danialeresbar As Francisco pointed out, this still needs to be updated.

File: djangoplicity/reports/models.py (hunk 5 above)
Comment (EdisonArango) [reply djangoplicity/reports/models.py]: @danialeresbar Please update this.

File: djangoplicity/metadata/archives/fields.py [hunk 8]
```diff
@@ -720,6 +720,7 @@ def _field_init( self, *args, **kwargs ):
             'blank': True,
             'help_text': AVMFacilityField.avm_help_text,
             'verbose_name': _( "Facility" ),
+            # 'on_delete': models.CASCADE
```
Comment (EdisonArango) [inline djangoplicity/metadata/archives/fields.py]: Please remove these comments, they are not necessary.

File: djangoplicity/science/migrations/0001_initial.py (hunk 6 above)
Comment (danialeresbar) [reply djangoplicity/science/migrations/0001_initial.py]: Ok

File: djangoplicity/science/models.py (hunk 7 above)
Comment (danialeresbar) [reply djangoplicity/science/models.py]: Well. It's done

**Pull request merged**